tests/test_p4_revert_unshelved.py

```diff
@@ -131,3 +133,38 @@ def create_file(filename):
 
     for result, expected in zip(diff, expected_path):
         assert result == expected
+
+
+@pytest.fixture()
+def perforce_environment(perforce_workspace, tmpdir):
+    yield P4Environment(perforce_workspace, tmpdir, test_type="main")
+
+
+def test_p4_error_revert(perforce_environment, stdout_checker, capsys):
+    p4 = perforce_environment.p4
+    p4_file = perforce_environment.repo_file
+
+    config = """
+from _universum.configuration_support import Variations
+
+configs = Variations([dict(name="Restrict changes", command=["chmod", "-R", "555", "."]),
+                      dict(name="Check", command=["ls", "-la"])])
+"""
+    p4.run_edit(perforce_environment.depot)
+    p4_file.write(config)
+    change = p4.fetch_change()
+    change["Description"] = "CL for shelving"
+    shelve_cl = p4.save_change(change)[0].split()[1]
+    p4.run_shelve("-fc", shelve_cl)
+
+    settings = perforce_environment.settings
+    settings.Launcher.output = "console"
+    settings.PerforceMainVcs.shelve_cls = [shelve_cl]
+    settings.Launcher.config_path = p4_file.basename
+
+    result = universum.run(settings)
+    perforce_environment.directory.chmod(0777, rec=1)
+    perforce_environment.directory.remove(rec=1)
+    assert result != 0
+    stdout_checker.assert_has_calls_with_param("Errors during command execution( \"p4 revert //...\" )")
+    assert "[Errno 13] Permission denied" in capsys.readouterr().err
```

Comment:
There is no check of correct handling the p4 revert exception, only the check for the error message.